honeycomb-render/src/representations/intermediates.rs

```diff
@@ -0,0 +1,31 @@
+use honeycomb_core::{CoordsFloat, Vertex2};
+
+pub enum Entity {
+    Dart,
+    Face,
+}
+
+pub struct IntermediateFace<T: CoordsFloat> {
+    pub vertices: Vec<Vertex2<T>>,
+    pub n_vertices: usize,
+    pub center: Vertex2<T>,
+}
+
+impl<T: CoordsFloat> IntermediateFace<T> {
+    pub fn new<I: Iterator<Item = Vertex2<T>>>(it: I) -> Self {
+        let tmp: Vec<Vertex2<T>> = it.collect();
+        let n_vertices = tmp.len();
+        let center = Vertex2::from(
+            tmp.iter()
+                .map(|vertex2: &Vertex2<T>| vertex2.into_inner())
+                .reduce(|coords1, coords2| coords1 + coords2)
+                .unwrap()
+                / T::from(n_vertices).unwrap(),
```

Comment:
`from` can fail or is it the division?
You can early exit if `n_vertices` is 0.

Reply:
It's the `from` call that can fail. I still added a `match` block in case the face is made up of only 0 or 1 vertex. These case will naturally yield an empty iterator when attempting to render later in the execution flow.